### PR TITLE
tests: Implement network delay and blackholing in linearizability tests

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @ptabor

### CODE REVIEW

File: tests/linearizability/failpoints.go [hunk 1]
```diff
@@ -56,6 +56,8 @@ var (
 	CompactBeforeCommitBatchPanic            Failpoint = goPanicFailpoint{"compactBeforeCommitBatch", triggerCompact, AnyMember}
 	CompactAfterCommitBatchPanic             Failpoint = goPanicFailpoint{"compactAfterCommitBatch", triggerCompact, AnyMember}
 	RaftBeforeLeaderSendPanic                Failpoint = goPanicFailpoint{"raftBeforeLeaderSend", nil, Leader}
+	BlackholePeerNetwork                     Failpoint = blackholePeerNetworkFailpoint{duration: time.Second}
```
Comment (lavacat) [inline tests/linearizability/failpoints.go:59]: nit: have you considered adding `target` (Leader, Follower, Quorum, All)? Similar to current functional test [cases](https://github.com/etcd-io/etcd/blob/main/tests/functional/rpcpb/rpc.proto#L431)

Comment (serathius) [reply tests/linearizability/failpoints.go:59]: For now I focused on getting proxy integrated into e2e. There are definitely couple of important cases in network functional failpoints that we should implement. Will add it to https://github.com/etcd-io/etcd/issues/14045

**Review by ptabor — APPROVED** [review]

File: tests/linearizability/failpoints.go [hunk 2]
```diff
@@ -247,3 +251,45 @@ func (f randomFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.Et
...
+	duration time.Duration
+}
+
+func (f blackholePeerNetworkFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error {
+	member := clus.Procs[rand.Int()%len(clus.Procs)]
+	proxy := member.PeerProxy()
+
+	proxy.BlackholeTx()
+	proxy.BlackholeRx()
+	time.Sleep(f.duration)
```
[the remark this answers is not in the document]
Comment (serathius) [reply tests/linearizability/failpoints.go:265]: Done

File: tests/linearizability/failpoints.go [hunk 3]
```diff
@@ -247,3 +251,45 @@ func (f randomFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.Et
...
+	randomizedLatency time.Duration
+}
+
+func (f delayPeerNetworkFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.EtcdProcessCluster) error {
+	member := clus.Procs[rand.Int()%len(clus.Procs)]
+	proxy := member.PeerProxy()
+
+	proxy.DelayRx(f.baseLatency, f.randomizedLatency)
+	proxy.DelayTx(f.baseLatency, f.randomizedLatency)
+	time.Sleep(f.duration)
```
[the remark this answers is not in the document]
Comment (serathius) [reply tests/linearizability/failpoints.go:287]: Done

**Pull request merged**